builtin/logical/ssh/path_roles.go

```diff
@@ -0,0 +1,245 @@
+package ssh
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathRoles(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "roles/(?P<name>[-\\w]+)",
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Name of the role",
+			},
+			"key": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Named key in Vault",
+			},
+			"admin_user": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Admin user at target address",
+			},
+			"default_user": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Default user to whom the dynamic key is installed",
+			},
+			"cidr": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "CIDR blocks and IP addresses",
+			},
+			"port": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Port number for SSH connection",
+			},
+			"key_type": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "one-time-password or dynamic-key",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathRoleRead,
+			logical.WriteOperation:  b.pathRoleWrite,
+			logical.DeleteOperation: b.pathRoleDelete,
+		},
+
+		HelpSynopsis:    pathRoleHelpSyn,
+		HelpDescription: pathRoleHelpDesc,
+	}
+}
+
+func createOTPRole(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	roleName := d.Get("name").(string)
+	if roleName == "" {
+		return logical.ErrorResponse("Missing role name"), nil
+	}
+
+	cidr := d.Get("cidr").(string)
+	if cidr == "" {
+		return logical.ErrorResponse("Missing cidr blocks"), nil
+	}
+	for _, item := range strings.Split(cidr, ",") {
+		_, _, err := net.ParseCIDR(item)
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("Invalid cidr entry '%s'", item)), nil
+		}
+	}
+
+	adminUser := d.Get("admin_user").(string)
+	defaultUser := d.Get("default_user").(string)
+	if defaultUser == "" && adminUser != "" {
```

Comment:
Ah, let's still be explicit and prevent an admin_user (error if provided) but require the default_user.